elastic/kibana
# [Security Solution][Execution Context] - Adds execution context to top pages for APM logs

## Summary

Addresses https://github.com/elastic/kibana/issues/126799

This PR introduces the client side ExecutionContextService within Security Solution. This utility is added to provide application context of client and server side interactions that are then passed on to APM.

This PR concentrates on high level routes within security solution. Individual teams may decide to on further granularity following this guidance - https://github.com/elastic/kibana/issues/126799

### Testing

- Enable APM by running export ELASTIC_APM_ACTIVE=true
- Run Kibana locally - turn off any ad blockers (they block FE events)
- Visit [discover](https://kibana-cloud-apm.elastic.dev/app/discover) and query something like `labels.kibana_uuid : "[YOUR UUID]" and labels.name: "securitySolutionUI"`
- If your uuid is statically set by ftr tests to 5b2de169-2785-441b-ae8c-186a1936b17d - delete your .uuid file.
- Navigate through the security solution pages and verify that you are able to see new logs reporting front end and backend events in the pages you visited

### Exceptions
<img width="2224" alt="Screen Shot 2022-04-13 at 2 18 49 PM" src="https://user-images.githubusercontent.com/10927944/163274341-2b1e0d47-d599-4ebd-bd11-ae89f2c615bb.png">

### Rules Create
<img width="2231" alt="Screen Shot 2022-04-13 at 2 19 15 PM" src="https://user-images.githubusercontent.com/10927944/163274389-c3a3ede1-b12a-47f9-992a-ab8fb7d77415.png">

### Rule Edit
<img width="2195" alt="Screen Shot 2022-04-13 at 2 19 47 PM" src="https://user-images.githubusercontent.com/10927944/163274418-f682a694-a901-40c2-80b2-19d9ed7ee0af.png">

### Rule Details 
<img width="2212" alt="Screen Shot 2022-04-13 at 2 19 35 PM" src="https://user-images.githubusercontent.com/10927944/163274517-bdcc734d-913b-4cd9-9dc0-313abc1f8ff8.png">

### Rules Management
<img width="2211" alt="Screen Shot 2022-04-13 at 2 20 00 PM" src="https://user-images.githubusercontent.com/10927944/163274554-e7f9e22c-afde-4e67-b002-814bcc556631.png">
